vnext/Microsoft.ReactNative/Views/DevMenu.cpp

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DevMenu.h"
+#include "Modules/DevSettingsModule.h"
+#include "UI.Xaml.Controls.Primitives.h"
+#include "UI.Xaml.Controls.h"
+#include "UI.Xaml.Input.h"
+#include "UI.Xaml.Interop.h"
+#include "UI.Xaml.Markup.h"
+#include "Utils/Helpers.h"
+#include "IReactDispatcher.h"
+#include "winrt/Windows.UI.Core.h"
+#include "Views/KeyboardEventHandler.h"
+
+namespace Microsoft::ReactNative {
+
+React::ReactPropertyId<React::ReactNonAbiValue<std::shared_ptr<DevMenuManager>>> DevMenuManagerProperty() noexcept
+{
+  static React::ReactPropertyId<React::ReactNonAbiValue<std::shared_ptr<DevMenuManager>>> propId {
+      L"ReactNative.DevMenuManager", L"DevMenuManager" };
+  return propId;
+}
+
+/*static*/ void DevMenuManager::InitDevMenu(Mso::CntPtr<Mso::React::IReactContext> const &reactContext) noexcept
+{
+  auto devMenu = std::make_shared<DevMenuManager>(reactContext);
+  devMenu->Init();
+  React::ReactPropertyBag(reactContext->Properties()).Set(DevMenuManagerProperty(), devMenu);
+}
+
+void DevMenuManager::Init() noexcept {
+  auto uiDispatcher = React::implementation::ReactDispatcher::GetUIDispatcher(m_context->Properties());
+  uiDispatcher.Post([weakThis = weak_from_this()]() {
+    if (auto pThis = weakThis.lock()) {
+      auto context = pThis->m_context;
+
+      if (react::uwp::Is19H1OrHigher()) {
+        // XamlRoot added in 19H1
+        if (auto xamlRoot = React::XamlUIService::GetXamlRoot(pThis->m_context->Properties())) {
+          if (auto rootContent = xamlRoot.Content()) {
+            pThis->m_keyDownRevoker = rootContent.KeyDown(
+                winrt::auto_revoke, [context](const auto & /*sender*/, const xaml::Input::KeyRoutedEventArgs args) {
+                  if (args.Key() == winrt::Windows::System::VirtualKey::D &&
+                      react::uwp::KeyboardHelper::IsModifiedKeyPressed(
+                          winrt::CoreWindow::GetForCurrentThread(), winrt::Windows::System::VirtualKey::Shift) &&
+                      react::uwp::KeyboardHelper::IsModifiedKeyPressed(
+                          winrt::CoreWindow::GetForCurrentThread(), winrt::Windows::System::VirtualKey::Control)) {
```

Comment:
nit: Perhaps a helper function `IsCtrlShiftD(winrt::Windows::System::VirtualKey key)` to share between 19h1 and before?